app/api/ms_api_utils.py

```diff
@@ -19,21 +15,65 @@
 # you run the MS API local server when you push the PR.
 # BASE_MS_API_URL = "https://bridge-in-tech-ms-test.herokuapp.com"
 
-
 # create instance
 def post_request(request_url, data):
+    
     try:
-
         response = requests.post(
             request_url, json=data, headers={"Accept": "application/json"}
         )
         response.raise_for_status()
-    except requests.ConnectionError as e:
-        return f"{e.response.json()}", HTTPStatus.INTERNAL_SERVER_ERROR
-    except requests.HTTPError as e:
-        return f"{e.response.json()}", e.response.status_code
+        response_message = response.json()
+        response_code = response.status_code
+    except requests.exceptions.ConnectionError as e:
+        response_message = messages.INTERNAL_SERVER_ERROR
+        response_code = json.dumps(HTTPStatus.INTERNAL_SERVER_ERROR)
+        logging.fatal(f"{e}")
+    except requests.exceptions.HTTPError as e:
+        response_message = e.response.json()
+        response_code = e.response.status_code
     except Exception as e:
-        return f"{e.response.json()}", e.response.status_code
-    else:
-        logging.warning(f"{response}")
-        return f"{response.json()}", response.status_code
+        response_message = messages.INTERNAL_SERVER_ERROR
+        response_code = json.dumps(HTTPStatus.INTERNAL_SERVER_ERROR)
+        logging.fatal(f"{e}")
+    finally:
+        logging.fatal(f"{response_message}")
+        return response_message, response_code
+    
+def http_response_checker(result, request_type):
+    
+    if result[1] == HTTPStatus.OK:
+        return http_ok_status_checker(result, request_type)
+    if result[1] == HTTPStatus.BAD_REQUEST:
+        return http_bad_request_checker(result)
+    
+    # TO DO. CHANGE HTTPStatus.NOT_FOUND to UNAUTHORIZED AFTER PR647 IN MS BACKEND IS APPROVED
+    if request_type == "login":
+        if result[1] == HTTPStatus.NOT_FOUND:
+            return messages.WRONG_USERNAME_OR_PASSWORD, HTTPStatus.UNAUTHORIZED
+    
+    if result[1] == HTTPStatus.FORBIDDEN:
+        return messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN, HTTPStatus.FORBIDDEN
+    if result[1] == HTTPStatus.INTERNAL_SERVER_ERROR:
+        return messages.INTERNAL_SERVER_ERROR, HTTPStatus.INTERNAL_SERVER_ERROR
+    # for all other errors
+    return result
+    
+def http_bad_request_checker(result):
+    
+    if result[0] == f"{messages.USERNAME_FIELD_IS_MISSING}":
```

Comment:
Won't this change now? should be indexed using the named tuple?

Reply:
Yes, It is now (on PR #65), already using namedtuple. This Login PR #60 was made before that namedtuple PR. I don't think I need to change in this PR since it'll be overwritten by PR #65 changes if that one also got approved after this.

screenshot from namedtuple ms_api_utils.py

<img width="689" alt="Screen Shot 2020-06-27 at 7 35 48 pm" src="https://user-images.githubusercontent.com/29667122/85919382-f5beda00-b8ad-11ea-8930-2c3df63ce09d.png">

The commit history showing sequence of changes

<img width="857" alt="Screen Shot 2020-06-27 at 7 36 02 pm" src="https://user-images.githubusercontent.com/29667122/85919392-0a9b6d80-b8ae-11ea-90d5-e65a410fdc74.png">